providers/desec/protocol.go

```diff
@@ -175,6 +176,17 @@ retry:
 	if resp.StatusCode > 299 {
 		if resp.StatusCode == 429 && retrycnt < 5 {
 			retrycnt++
+			//we've got rate limiting and will try to get the Retry-After Header if this fails we fallback to sleep for 500ms max. 5 retries.
+			waitfor := resp.Header.Get("Retry-After")
+			if waitfor != "" {
+				wait, err := strconv.ParseInt(waitfor, 10, 64)
+				if err == nil {
+					printer.Warnf("Rate limiting.. waiting for %s seconds", waitfor)
+					time.Sleep(time.Duration(wait) * time.Second)
+					goto retry
+				}
+			}
+			printer.Warnf("Rate limiting.. waiting for 500 milliseconds")
```

Comment:
the message is incorrect

Reply:
No its not since this is the fallback, if it can't get the Retry-After header it will wait for 500 milliseconds as it did before.
But if everything works as expected this code and message should never be triggered anymore.

Reply:
My bad, I didn't look thoroughly enough. Sorry!